streams/src/main/java/org/apache/kafka/streams/processor/TimestampExtractor.java

```diff
@@ -26,10 +26,10 @@
 public interface TimestampExtractor {
 
     /**
-     * Extracts a timestamp from a message
+     * Extracts a timestamp from a message.
```

Comment:
Should be sth like:

> Extracts a timestamp from a record.
>
> Typically, the timestamp should represent the milliseconds since midnight, 1 January 1970 UTC.

@guozhangwang : Should we say "should" or "must" represent the milliseconds since the epoch?  KIP-32, for example, explicitly defines the contract to be milliseconds since the epoch, though in the context Streams there's nothing that would specifically force the user to stick to that definition -- a custom timestamp extractor could, for example, simple return seconds since 1 April 2000, and the Streams app would still work (as long as you are not joining such a stream with a stream that has a different interpretation/semantically different extraction of time).